rs/tests/ict/cmd/helpers.go

```diff
@@ -149,3 +151,101 @@ func get_closest_target_matches(all_targets []string, target string) []string {
 		return len(s) > 0
 	})
 }
+
+func sparse_checkout(repoUrl, repoDir string, sparseCheckoutPaths []string) (string, error) {
+	startingPoint, err := os.Getwd()
+	if err != nil {
+		return "", fmt.Errorf("Could not get current dir: %v", err)
+	}
+
+	if repoDir == "" {
+		homedir, err := os.UserHomeDir()
+		if err != nil {
+			return "", fmt.Errorf("Failed to get home dir: %v", err)
+		}
+		repoDir = filepath.Join(homedir, ".cache", "k8s_repo")
+	}
+
+	if err := os.RemoveAll(repoDir); err != nil {
+		return return_to_starting_point(startingPoint, fmt.Errorf("Failed to remove directory: %v", err))
```

Comment:
would it make sense to use `defer` instead of always having to make sure this is called?

```go
func foo() {
	returnToStartingPoint := true
	defer func() {
		if returnToStartingPoint {
			// ...
		} 
        }
        ...
        returnToStartingPoint = false;
}